backends-velox/src/test/scala/org/apache/gluten/execution/ScalarFunctionsValidateSuite.scala

```diff
@@ -148,7 +148,7 @@ abstract class ScalarFunctionsValidateSuite extends FunctionsValidateSuite {
     }
   }
 
-  testWithSpecifiedSparkVersion("Test array_append function - STRING", Some("3.4")) {
+  testAtLeastSparkVersion("Test array_append function - STRING", "3.4") {
```

Comment:
If we do this refactor, can we make arguments in `testWithSpecifiedSparkVersion` non-optional? So user have clearer choice to use only one API at a time.